src/Processor/UpdatedOrderProcessor.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusOrderEditPlugin\Processor;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Setono\SyliusOrderEditPlugin\Exception\NewOrderWrongTotalException;
+use Sylius\Component\Core\Inventory\Operator\OrderInventoryOperatorInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\Processor\OrderProcessorInterface;
+
+final class UpdatedOrderProcessor
+{
+    public function __construct(
+        private OrderProcessorInterface $orderProcessor,
+        private OrderInventoryOperatorInterface $orderInventoryOperator,
+        private OrderProcessorInterface $afterCheckoutOrderPaymentProcessor,
+        private EntityManagerInterface $entityManager,
```

Comment:
I never inject the entity manager. I use this library for these purposes: https://github.com/Setono/doctrine-orm-trait

Reply:
Interesting - any particular reason for such an approach?

Reply:
Haha, not a reason that wins an argument 😂 I guess Matthias has a few here: https://matthiasnoback.nl/2014/05/inject-the-manager-registry-instead-of-the-entity-manager/, but also here are a few of my own
- I just got accustomed doing it
- The entity manager can be closed and you can reset it
- You _know_ that no matter what magic people have done, you will get the right manager
- I get some typing from Psalm that Sylius doesn't provide yet

As I said, nothing that wins arguments :D

Reply:
Let me then pause it for now to not make the iteration over this PR longer than necessary, but we can definitely come back to this discussion later, ok? :)

Reply:
Yes, of course. Let's move on